### PR TITLE
[9.0][ADD] Forum Censorship

### PR DESCRIPTION
**Pull request by tarteo**
Enables you to censor phrases on forums.

### CODE REVIEW

File: website_forum_censored/models/forum_post.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @api.model
+    def create(self, values):
+        phrases = self.env['forum.censored.phrase'].search([])
+        for phrase in phrases:
+            ptn = compile(phrase.phrase, IGNORECASE)
+            if 'name' in values:
+                values['name'] = ptn.sub(phrase.replacement, values['name'])
+            if 'content' in values:
+                values['content'] = ptn.sub(phrase.replacement,
+                                            values['content'])
```
Comment (gurneyalex) [inline website_forum_censored/models/forum_post.py:21]: from a stylistic point of view, I'd have a method `replace()` on the `forum.censored.phrase` model.

File: website_forum_censored/models/forum_censored_phrase.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# © 2016 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models
+
+
+class ForumCencoredPhrase(models.Model):
+    _name = 'forum.censored.phrase'
+
+    phrase = fields.Char('Phrase', required=True)
```
Comment (gurneyalex) [inline website_forum_censored/models/forum_censored_phrase.py:13]: you should validate this. It is easy to create a phrase which is not a valid regexp, which will break with an obscure message on the creation of a forum post. I'd rather have a python constraint on the `phrase` field which makes sure the expression can be compiled.

File: website_forum_censored/README.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+* Add option to obstruct creation of forum post.
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/website/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
```
Comment (elicoidal) [inline website_forum_censored/README.rst]: you can remove this line (check newest template)

**Comment by tarteo** [comment]
@gurneyalex Thanks for the review! I'll fix it asap.

**Comment by ehahouimohssine** [comment]
:+1:

**Comment by elicoidal** [comment]
:+1:

**Comment by gurneyalex** [comment]
:+1: Could you squash your commits?

**Review by rafaelbn — APPROVED** [review]

**Pull request merged**